### PR TITLE
Add an ec2 utility function for getting AMI Info.

### PR DESCRIPTION
**Pull request by feanil**
The edc_for_ami function will get the EDC tags of an AMI using boto.

### CODE REVIEW

**Comment by feanil** [comment]
@macdiesel @doctoryes This bit is stand-alone and working so I thought I would get it reviewed independently and merge as a beginning for the code migration work.

File: tubular/ec2.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+    try:
+        ami = ec2.get_all_images(ami_id)[0]
+    except EC2ResponseError as error:
+        raise ImageNotFoundException(error.message)
+
+    tags = ami.tags
+
+    # TODO How do we want to handle these tags not existing?
+    # raise an exception maybe? Right now this is not safe.
```
Comment (macdiesel) [inline tubular/ec2.py]: nit: It does raise an exception currently, should update Doc

File: tubular/ec2.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+def edc_for_ami(ami_id):
+    """
+    Look up the EDC tags for an AMI.
+
+    Input: An AMI Id.
+    Output: An EDC Named Tuple.
+    Exceptions:
+        - ImageNotFoundException
+        - MissingTagException
+    """
```
Comment (macdiesel) [inline tubular/ec2.py:25]: Docstrings should match edx recommended style:
https://github.com/edx/edx-platform/wiki/Python-Guidelines (assuming this is still current)

File: tubular/utils.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from collections import namedtuple
+
+class TimeoutException(Exception):
```
Comment (macdiesel) [inline tubular/utils.py]: What do you think about putting exceptions in their own Exceptions.py file? This seems a bit out of place.

**Comment by macdiesel** [comment]
Done with my pass.

File: tubular/utils.py (hunk 3 above)
Comment (feanil) [inline tubular/utils.py]: Seems reasonable, would the ones currently in ec2.py also go there?

**Comment by feanil** [comment]
@macdiesel comments addressed. I decided to move all the exceptions into the new exceptions module since I'm seeing a few libraries I'm using already using that pattern.

**Comment by macdiesel** [comment]
:ship: :it: :+1:

**Pull request merged**